backend/eventloop/repocred_reconciler_test.go

```diff
@@ -90,39 +91,123 @@ var _ = Describe("RepoCred Reconcile Function Tests", func() {
 
 			err = dbq.CreateAPICRToDatabaseMapping(ctx, &apiCRToDatabaseMappingDb)
 			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should set an error status for RepositoryCredentials if the secret field is not set in the CR", func() {
+
+			defer dbq.CloseDatabase()
 
+			By("creating a GitOpsDeployment CR in cluster")
 			gitopsDeploymentRepositoryCredentialCR := &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      apiCRToDatabaseMappingDb.APIResourceName,
 					Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
 				},
 			}
-			// Create GitOpsDeployment CR in cluster
-			err = k8sClient.Create(context.Background(), gitopsDeploymentRepositoryCredentialCR)
+			err := k8sClient.Create(context.Background(), gitopsDeploymentRepositoryCredentialCR)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("calling the Reconcile function.")
+			reconcileRepositoryCredentials(ctx, dbq, k8sClient, log)
 
+			By("verifing that status is updated for GitopsDeploymentRepositoryCredentialCR.")
+			objectMeta := metav1.ObjectMeta{
+				Name:      apiCRToDatabaseMappingDb.APIResourceName,
+				Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
+			}
+			repoCredCR := &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{ObjectMeta: objectMeta}
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(repoCredCR), repoCredCR)
 			Expect(err).ToNot(HaveOccurred())
+			Expect(repoCredCR).NotTo(BeNil())
+			Expect(repoCredCR.Status.Conditions).To(HaveLen(3))
+			Expect(repoCredCR.Status.Conditions[0].Type).To(Equal(managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred))
+			Expect(repoCredCR.Status.Conditions[0].Reason).To(Equal(managedgitopsv1alpha1.RepositoryCredentialReasonSecretNotSpecified))
+			Expect(repoCredCR.Status.Conditions[0].Status).To(Equal(metav1.ConditionTrue))
+			Expect(repoCredCR.Status.Conditions[0].Message).To(Equal("Secret field is missing value"))
```

Comment:
Minor tweak: rather than expecting the condition to be a position 0, I presume we can search for the condition with the specific Type. 

(this just makes this test slightly less brittle if we happen to unintentionally change the order in the condition generation function.)

Reply:
Done